orc8r/gateway/python/magma/magmad/BUILD.bazel

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@python_deps//:requirements.bzl", "requirement")
+load("@rules_python//python:defs.bzl", "py_binary")
+
+ORC8R_ROOT = "../../"
+
+DEPS = [
+    "//orc8r/gateway/python/magma/common:grpc_client_manager",
+    "//orc8r/gateway/python/magma/common:sdwatchdog",
+    "//orc8r/gateway/python/magma/common:sentry",
+    "//orc8r/gateway/python/magma/common:cert_utils",
+    "//orc8r/gateway/python/magma/common:cert_validity",
+    "//orc8r/gateway/python/magma/common:rpc_utils",
+    "//orc8r/gateway/python/magma/common:misc_utils",
+    "//orc8r/gateway/python/magma/common:streamer",
+    "//orc8r/gateway/python/magma/common:stateless_agw",
+    "//orc8r/gateway/python/magma/common/health:service_state_wrapper",
+    "//orc8r/gateway/python/magma/magmad/check/kernel_check:kernel_versions",
+    "//orc8r/gateway/python/magma/magmad/check/machine_check:cpu_info",
+    "//orc8r/gateway/python/magma/magmad/check/network_check:ping",
+    "//orc8r/gateway/python/magma/magmad/check/network_check:routing_table",
+    "//orc8r/gateway/python/magma/magmad/generic_command:command_executor",
+    "//orc8r/gateway/python/magma/magmad/upgrade:upgrader",
+    "//orc8r/gateway/python/magma/magmad/upgrade:magma_upgrader",
+    "//orc8r/protos:bootstrapper_python_grpc",
+    "//orc8r/protos:metricsd_python_grpc",
+    "//orc8r/protos:state_python_grpc",
+    "//orc8r/protos:magmad_python_grpc",
+    "//orc8r/protos:sync_rpc_service_python_grpc",
+    "//orc8r/swagger:magmad_events_v1",
+    "@aioh2_repo//:aioh2",
+    requirement("snowflake"),
+    requirement("netifaces"),
+]
+
+SRCS = [
+    "main.py",
+    "bootstrap_manager.py",
+    "config_manager.py",
+    "events.py",
+    "gateway_status.py",
+    "metrics_collector.py",
+    "metrics.py",
+    "proxy_client.py",
+    "rpc_servicer.py",
+    "service_health_watchdog.py",
+    "service_manager.py",
+    "service_poller.py",
+    "state_reporter.py",
+    "sync_rpc_client.py",
+]
+
+py_binary(
+    name = "magmad",
+    srcs = SRCS,
+    imports = [ORC8R_ROOT],
+    # legacy_creat_init = False is required to fix issues in module import, see https://github.com/rules-proto-grpc/rules_proto_grpc/issues/145
+    legacy_create_init = False,
+    main = "main.py",
+    python_version = "PY3",
+    visibility = ["//visibility:public"],
+    deps = DEPS,
```

Comment:
A general comment with no specific solution in mind.. 

This large list of dependency worries me that it'll be harder for people to keep these up to day. I'm thinking of cases where we remove dependencies from the code but we forget to remove dependencies from the build file. I was wondering if creating smaller `py_library`s would help, but I am not sure. What is your opnion here on helping devs manage these in the long run?

Reply:
Force pushed the main.py/library approach, the dependencies are now split somewhat more. What is your opinion on this? We suggest to use this approach as a coding pattern for future services.